### PR TITLE
Precise compiler invocations: Set C++ interoperability per SymbolContext 

### PR DESCRIPTION
**Pull request by adrian-prantl**
rdar://125182583

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by m7mdragab — APPROVED** [review]

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -2172,6 +2172,41 @@ SwiftASTContext::CreateInstance(lldb::LanguageType language, Module &module,
...
+    ModuleList::GetGlobalModuleListProperties().GetSwiftEnableCxxInterop();
+  switch (interop_enabled) {
+  case AutoBool::True:
+    return true;
+  case AutoBool::False:
+    return false;
+  case AutoBool::Auto: {
+    if (!cu)
+      return false;
+    lldb::ModuleSP module = cu->CalculateSymbolContextModule();
```
Comment (augusto2112) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:2187]: Why can't you call `Module::IsSwiftCxxInteropEnabled`? The code looks pretty much the same.

**Review by augusto2112 — CHANGES_REQUESTED** [review]

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 2]
```diff
@@ -2172,6 +2172,41 @@ SwiftASTContext::CreateInstance(lldb::LanguageType language, Module &module,
...
+    if (!module)
+      return false;
+    // Look for the "-enable-experimental-cxx-interop" compile flag in the
+    // args of the compile units this module is composed of.
+    auto *sym_file = module->GetSymbolFile();
+    if (!sym_file)
+      return false;
+    auto options = sym_file->GetCompileOptions();
+    for (auto &[unit, args] : options) {
+      if (unit.get() == cu) {
```
Comment (adrian-prantl) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:2197]: Because of this check.

Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp:2197]: @augusto2112 ^

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Review by augusto2112 — APPROVED** [review]

**Pull request merged**